testing.py

```diff
@@ -233,3 +234,39 @@ def zip_11212(self):
 
     zip_11212_geojson = '{"type": "Polygon", "coordinates": [[[-73.931607, 40.663557], [-73.930685, 40.66362], [-73.928544, 40.664574], [-73.92638, 40.665515], [-73.924389, 40.666385], [-73.922753, 40.667098], [-73.918382, 40.669002], [-73.920044, 40.669137], [-73.914873, 40.670533], [-73.903475, 40.675507], [-73.900417, 40.663806], [-73.900661, 40.660587], [-73.898807, 40.657407], [-73.899041, 40.657372], [-73.903173, 40.655301], [-73.907856, 40.654589], [-73.909726, 40.654313], [-73.917576, 40.654314], [-73.921005, 40.654918], [-73.921776, 40.654424], [-73.927201, 40.658537], [-73.927255, 40.659338], [-73.928193, 40.660177], [-73.931556, 40.662756], [-73.931607, 40.663557]]]}'
+
+
+class DummyHTTPResponse(object):
+    def __init__(self, status_code, headers, content):
+        self.status_code = status_code
+        self.headers = headers
+        self.content = content
+
+class DummyHTTPClient(object):
+
+    def __init__(self):
+        self.responses = []
+        self.requests = []
+
+    def queue_response(self, response_code, media_type="text/html",
+                       other_headers=None, content=''):
+        headers = {}
+        if media_type:
+            headers["content-type"] = media_type
+        if other_headers:
+            for k, v in other_headers.items():
+                headers[k.lower()] = v
+        self.responses.append(DummyHTTPResponse(response_code, headers, content))
+
+    def do_get(self, url, headers=None, allowed_response_codes=None, **kwargs):
+        self.requests.append(url)
+        response = self.responses.pop()
+
+        code = response.status_code
+        series = "%sxx" % (code / 100)
+
+        if allowed_response_codes and (
+            code not in allowed_response_codes and series not in allowed_response_codes
+            ):
+            raise BadResponseException(url, "Bad Response!", status_code=code)
+        return response
```

Comment:
This is brought over from core, but I changed it so I could use it in place of HTTP.get_with_timeout instead of having to define another method that calls that.